### PR TITLE
Create VT extensions list for documentation

### PR DESCRIPTION
**Pull request by Yaraslaut**
Generate vt sequence documentation for web page

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -19,12 +19,21 @@ jobs:
...
           ./scripts/extract-changelog-from-metainfo.py metainfo.xml docs/release-notes.md
 
+      - name: "install dependencies"
+        run: sudo env QTVER=6 SYSDEP_ASSUME_YES=ON ./scripts/install-deps.sh
+      - name: Install-contour
+        run: cmake --preset linux-debug -B build -G Ninja
+      - name: "build"
+        run: cmake --build build
+      - name: "Generate vt documentation"
+        run: ./build/src/contour/contour documentation vt > docs/vt-extensions/index.md
```
Comment (christianparpart) [inline .github/workflows/docs.yml:35]: I think extensions is the wrong place, or is it? It contains most standard ones plus some extensions.... Hmm.....

Comment (Yaraslaut) [reply .github/workflows/docs.yml:39]: Which place do you prefer?

Comment (christianparpart) [reply .github/workflows/docs.yml:39]: At first I thought internals. But vt sequence documentation targets power users and TUI/CLI devs... So what aber "advanced" user documentation? :thinking:

**Comment by christianparpart** [comment]
Hugh? You closed it?

**Comment by Yaraslaut** [comment]
> Hugh? You closed it?

by accident, pushed in wrong commit :( will reopen it soon

File: src/contour/ContourApp.cpp [hunk 2]
```diff
@@ -178,6 +179,45 @@ auto withOutput(crispy::cli::flag_store const& flags, std::string const& name, C
...
 }
 
+int ContourApp::documentationVT()
+{
+    using category = vtbackend::FunctionCategory;
+    using namespace std::string_view_literals;
+
+    std::string info;
+    auto back = std::back_inserter(info);
+    fmt::format_to(back, "#{}\n", "VT sequences");
```
Comment (christianparpart) [inline src/contour/ContourApp.cpp:189]:
```suggestion
    fmt::format_to(back, "# {}\n\n", "VT sequences");
```

File: src/contour/ContourApp.cpp [hunk 3]
```diff
@@ -178,6 +179,45 @@ auto withOutput(crispy::cli::flag_store const& flags, std::string const& name, C
...
+    fmt::format_to(back, "{}\n", "List of VT sequences supported by Contour Terminal Emulator.");
+    for (auto const& [category, headline]: { std::pair { category::C0, "Control Codes"sv },
+                                             std::pair { category::ESC, "Escape Sequences"sv },
+                                             std::pair { category::CSI, "Control Sequences"sv },
+                                             std::pair { category::OSC, "Operating System Commands"sv },
+                                             std::pair { category::DCS, "Device Control Sequences"sv } })
+    {
+
+        fmt::format_to(back, "## {}\n", headline);
```
Comment (christianparpart) [inline src/contour/ContourApp.cpp:198]:
```suggestion
        fmt::format_to(back, "## {}\n\n", headline);
```

File: src/contour/ContourApp.cpp [hunk 4]
```diff
@@ -178,6 +179,45 @@ auto withOutput(crispy::cli::flag_store const& flags, std::string const& name, C
...
 
+int ContourApp::documentationVT()
+{
+    using category = vtbackend::FunctionCategory;
+    using namespace std::string_view_literals;
+
+    std::string info;
+    auto back = std::back_inserter(info);
+    fmt::format_to(back, "#{}\n", "VT sequences");
+    fmt::format_to(back, "{}\n", "List of VT sequences supported by Contour Terminal Emulator.");
```
Comment (christianparpart) [inline src/contour/ContourApp.cpp:190]:
```suggestion
    fmt::format_to(back, "{}\n\n", "List of VT sequences supported by Contour Terminal Emulator.");
```

File: src/contour/ContourApp.cpp [hunk 5]
```diff
@@ -178,6 +179,45 @@ auto withOutput(crispy::cli::flag_store const& flags, std::string const& name, C
...
+        for (auto const& fn: vtbackend::allFunctions())
+        {
+            if (fn.category != category)
+                continue;
+
+            // This could be much more improved in good looking and informationally.
+            // We can also print short/longer description, minimum required VT level,
+            // colored output for easier reading, and maybe more.
+            fmt::format_to(back,
+                           "`{:}` {:} [`{:}`] <br/>\n",
```
Comment (christianparpart) [inline src/contour/ContourApp.cpp:209]: You might avoid the use of the HTML-like `<br/>` tag by going with the table approach. I think what MS did there looks quite nice: https://learn.microsoft.com/en-us/windows/console/console-virtual-terminal-sequences

This is possible for us, too. :)

Comment (Yaraslaut) [reply src/contour/ContourApp.cpp]: Update with tables

**Review by christianparpart — APPROVED** [review]
Thank you, @Yaraslaut :)

**Pull request merged**